packages/experimental/src/experimental-list/collapsible-list/index.tsx

```diff
@@ -78,6 +82,34 @@ function getUpdatedShownChildren(
 	);
 }
 
+const getTransitionStyle = (
+	state: 'entering' | 'entered' | 'exiting' | 'exited',
+	isCollapsed: boolean,
+	elementRef: HTMLDivElement | null
+) => {
+	let maxHeight = 0;
+	if ( ( state === 'entered' || state === 'entering' ) && elementRef ) {
+		maxHeight = getContainerHeight( elementRef );
+	}
+	const styles: React.CSSProperties = {
+		...defaultStyle,
+		maxHeight,
+	};
+
+	// only include transition styles when entering or exiting.
+	if ( state !== 'entering' && state !== 'exiting' ) {
+		delete styles.transitionDuration;
+		delete styles.transition;
+		delete styles.transitionProperty;
+	}
+	// Remove maxHeight when entered, so we do not need to worry about nested items changing height while expanded.
+	if ( state === 'entered' && ! isCollapsed ) {
+		delete styles.maxHeight;
+	}
```

Comment:
When an item is expanded and the transition finished (`entered`) we remove the maxHeight. This way if a nested item could be expanded we don't have to worry about calculating the height, as it will auto adjust).